packages/smithy-client/src/extensions/defaultClientConfiguration.ts

```diff
@@ -0,0 +1,30 @@
+import type { DefaultClientConfiguration } from "@smithy/types";
+
+import { getChecksumConfiguration, resolveChecksumRuntimeConfig } from "./checksum";
+
+/**
+ * @internal
+ */
+export type DefaultExtensionConfigType = Parameters<typeof getChecksumConfiguration>[0];
```

Comment:
This type replaces `any`. It can form a union later when more extensions exist. Is that consistent with the intended behavior?